File: C++/ValidSudoku.cpp
#include<iostream>
#include<stdlib.h>
#include<vector>
#include<unordered_set>
#include<string>
#include<string.h>
#include<algorithm>
#include<math.h>
#include<map>
using namespace std;
class Solution {
public:
	bool isValidSudoku(vector<vector<char> > &board) {
		unordered_set<char> unique;
		for(int i = 0; i < board.size(); i++)
		{
			unique.clear();
			for(int j = 0; j < board[i].size(); j++)
			{
				if(board[i][j] == '.')
					continue;
				if(unique.find(board[i][j]) != unique.end())
					return false;
				unique.insert(board[i][j]);
			}
			unique.clear();
			for(int j = 0; j < board.size(); j++)
			{
				if(board[j][i] == '.')
					continue;
				if(unique.find(board[j][i]) != unique.end())
					return false;
				unique.insert(board[j][i]);
			}
			unique.clear();
			int row = i/3;
			int col = i%3;
			for(int j = row*3; j < row*3+3; j++)
			{
				for(int k = col*3; k < col*3+3; k++)
				{
					if(board[j][k] == '.')
						continue;
					if(unique.find(board[j][k]) != unique.end())
						return false;
					unique.insert(board[j][k]);
				}
			}
		}
		return true;
	}
};

File: JAVA/ValidSudoku.java
import java.util.HashSet;
import java.util.Set;

class Solution {
    public boolean isValidSudoku(char[][] board) {
        // Start typing your Java solution below
        // DO NOT write main() function
        if(board == null || board.length != 9 || board[0].length != 9)
        	return false;
        Set<Integer> set = new HashSet<Integer>();

        for(int i = 0; i < 9; i++)
        {
        	set.clear();
        	for(int j = 0; j < 9; j++)
        	{
        		if(board[i][j] == '.')
        			continue;
        		else if(set.contains(board[i][j] - '0'))
        			return false;
        		set.add(board[i][j] - '0');
        	}
        	set.clear();
        	for(int j = 0; j < 9; j++)
        	{
        		if(board[j][i] == '.')
        			continue;
        		else if(set.contains(board[j][i] - '0'))
        			return false;
        		set.add(board[j][i] - '0');
        	}
        	set.clear();
        	int row = (i/3)*3;
        	int col = (i%3)*3;
        	for(int k = row; k < row + 3; k++)
        	{
        		for(int m = col; m < col + 3; m++)
        		{
            		if(board[k][m] == '.')
            			continue;
            		else if(set.contains(board[k][m] - '0'))
            			return false;
            		set.add(board[k][m] - '0');
        		}
        	}
        }
    	return true;
    }
}
public class ValidSudoku {

	/**
	 * @param args
	 */
	public static void main(String[] args) {
		// TODO Auto-generated method stub

	}

}
